src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -776,23 +776,19 @@ public byte[] ToArray()
 
             private void EnsureCapacity(int value)
             {
-                if (value > _buffer.Length)
+                if ((uint)value > (uint)_maxBufferSize) // value cast handles overflow to negative as well
                 {
-                    Grow(value);
+                    throw CreateOverCapacityException(_maxBufferSize);
                 }
-                else if (value < 0) // overflow
+                else if (value > _buffer.Length)
                 {
-                    throw CreateOverCapacityException(_maxBufferSize);
+                    Grow(value);
                 }
             }
 
             private void Grow(int value)
             {
                 Debug.Assert(value > _buffer.Length);
-                if (value > _maxBufferSize)
```

Comment:
Should we assert this?  Not sure if Grow is called anywhere else, but an Assert seems worthwhile.